src/algorithms/calorimetry/ImagingTopoCluster.h

```diff
@@ -198,7 +210,7 @@ namespace eicrecon {
     }
 
     // grouping function with Breadth-First Search
-    void bfs_group(const edm4eic::CalorimeterHitCollection &hits, std::set<std::size_t> &group, std::size_t idx, std::vector<bool> &visits) const {
+    void bfs_group(const edm4eic::CalorimeterHitCollection &hits, const std::vector<std::size_t>& indices, std::set<std::size_t> &group, std::size_t idx, std::vector<bool> &visits) const {
```

Comment:
`indices` as a set can serve as `visits`.
```suggestion
    void bfs_group(const edm4eic::CalorimeterHitCollection &hits, std::vector<std::size_t>& indices, std::set<std::size_t> &group, std::size_t idx) const {
```

Reply:
Implemented.